### DISCUSSION TITLE
Warning "Overriding id of vnode"

### DISCUSSION THREAD

**Discussion by MoHcmp** [eclipse-glsp/glsp#464] [Q&A]
When using glsp editor i get a lot of warnings with message "Overriding id of vnode (object-type_2_905e60ef-cb81-4aa6-b6fd-8ac42d91a56d). Make sure not to set it manually in view.". This warning comes from sprotty IdPostprocessor. What am i doing wrong?

**Comment by planger** [comment]
It looks like you assign an `id` attribute to the returned `VNode` in your views' `render` method, which the `IdPostprocessor` overwrites by re-assigning it to the respective `SModelElement` id.

**Reply by MoHcmp** [reply]
i do not assign `id` to `VNode`. I debugged this behaviour. `IdPostprocessor ` does it  when node first gets into `decorate` (it sets `attrs.id = this.domHelper.createUniqueDOMElementId(element);`, which is equal to id in warning). And when `VNode` gets into `decorate` method second time it argues that `id` was already set
